### ISSUE TITLE
Scaling Open-Vocabulary Object Detection

### ISSUE THREAD

**Issue by zhuoxiao-chen** [zhuoxiao-chen/please_read_paper_every_day#47]
## Summary

<img width="800" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/808cadf3-36e2-4b2c-8de1-2c8d02736ec0">

### Paper Link
https://arxiv.org/abs/2306.09683

## Motivation

**Image-level tasks**: large improvements have been achieved through contrastive pretraining of vision language
models, which is massively scalable because it can use naturally abundant weak supervision in the form of image-text pairs from the Web. However, no such natural supervision data exists for localisation tasks. 

**Self-training**: an existing detector is used to predict bounding boxes on unlabelled images to generate data for training better detectors. While existing work have done so at relatively small scales, comparable to the size of human-annotated detection datasets and much smaller than the datasets used for image-level training.

**Goal**: to scale self-training further for the object detection task. (title)

## Methodology

<img width="1573" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/6bd8e513-69ec-4d32-8e03-e8aa572004d9">

### Step 1: Generating Web-Scale Open-Vocabulary Object Annotations

- **WebLI dataset for self-training**: a large dataset of images and texts available on the public Web. The dataset consist of approximately 10B images and associated alt-text strings, which can be thought of as noisy image captions.
- **OWL-ViT CLIP-L/14** to annotate all 10B WebLI images with bounding box pseudo annotations. Given an image, the model first
detects objects in the image in a class-agnostic way. Then, given a list of free-text queries, the model produces scores indicating the likelihood that each detected object is associated with each text query. 
- Extract all word **N-grams** up to length 10 from the text associated with a given image and use them as queries for that image. Then apply minimal filtering, only removing generic terms like image or png, and queries consisting entirely of stop-words.

**Why N-grams**? Prior work using image captions as weak supervision for detection often used **grammatical parsing to extract noun phrases or concepts**. These approaches may add biases that reduce the diversity of extracted queries.

- Regardless of label space, we ensemble predictions over seven prompt templates such as "a photo of a {}".
- For each predicted box, we keep the query with the highest score as its pseudo-label. For each image, we keep all boxes above a score threshold.

### Step 2: Self-training at Scale

We optimise training efficiency to maximise the number of images seen for a given amount of training compute as follows:

1. **Token Dropping.** We drop tokens simply based on the **pixel variance** of the corresponding image patch. Both natural and Web images contain low-variance areas devoid of useful information, e.g. **sky, single-color backgrounds, or padding**.  We drop 50% of patches in all of our experiments during training. No patches are dropped during inference.

<img width="1332" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/bd6f48f7-e8f7-440a-8a5f-65b0f4ac6541">

2. **Instance selection.** OWL-ViT predicts **one bounding box per encoder token**. This is **inefficient**, since there are typically many more encoder tokens than objects (e.g. 5184 tokens for resolution 1008×1008 and patch size 14×14). **Most output tokens therefore do not represent objects**. We introduce an **objectness head** which predicts the likelihood that an output token actually represents an object, and compute boxes, class scores, and losses **only for the top k tokens** by objectness. We select approximately **10% of instances by top objectness** during training in all of our experiments. During inference, all instances are used.
3. **Mosaics**. We combine raw images into grids of up to 6 × 6 to produce a single training example. 1) increases the number of raw images. 2) Mosaics reduce the average object size and improve small-object performance. 

Our improvements reduce training FLOPS by approximately 50% compared to the original OWL-ViT and increase training throughput by 2×. We refer to the improved model as OWLv2. Note that at inference, the model is identical to the original OWL-ViT.

### Step 3: Fine-tuning

<img width="1050" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/aa5540f2-b294-4301-bfb5-cc641c4c51ac">

## Experiments 
<img width="996" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/31eb78f8-eb05-4575-b913-48d618c0ba15">

**Human-curated label space.**  Combinethe label sets from the LVIS [8], Objects365 [27], OpenImagesV4 [15], and Visual Genome [13] datasets and removing duplicates and plural forms. In total, this label space contains 2520 common object categories, e.g. "phone", "goatee", "teakettle", "park", "suit (clothing)". Note, LVIS_rare classes are still unseen for all of our models. 

<img width="1137" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/138518f5-62ba-4c2c-afe9-e573a634e782">

<img width="1425" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/92401f9f-f73c-4e34-8b75-1a2282cdf6ab">
